### PR TITLE
Ensure subp doesn't run during logs.py import

### PR DESCRIPTION
**Pull request by TheRealFalcon**
## Proposed Commit Message

```
See individual commits
```

## Additional Context
`tests/unittests/cmd/devel/test_logs.py` would fail if you ran the file on its own due to subp happening on import.

The 2nd commit is just some extra cleanup.

## Merge type

- [ ] Squash merge using "Proposed Commit Message"
- [x] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: cloudinit/cmd/devel/logs.py [hunk 1]
```diff
@@ -27,8 +27,23 @@
...
-CLOUDINIT_RUN_DIR = PATHS.run_dir
+
+class LogPaths(NamedTuple):
+    userdata_raw: str
+    cloud_data: str
+    run_dir: str
+    instance_data_sensitive: str
+
+
+def get_log_paths(init: Optional[Init] = None) -> LogPaths:
```
Comment (holmanb) [inline cloudinit/cmd/devel/logs.py:38]: +1 nice

File: cloudinit/cmd/main.py [hunk 2]
```diff
@@ -1050,7 +1050,7 @@ def main(sysv_args=None):
                 handle_collect_logs_args,
             )
 
-            logs_parser(parser_collect_logs)
+            logs_parser(parser=parser_collect_logs)
```
Comment (holmanb) [inline cloudinit/cmd/main.py:1053]: was this intended to be in the cleanup commit?

File: cloudinit/cmd/devel/logs.py [hunk 3]
```diff
@@ -293,7 +298,11 @@ def collect_logs(tarfile, include_userdata: bool, verbosity=0):
 
 def handle_collect_logs_args(name, args):
     """Handle calls to 'cloud-init collect-logs' as a subcommand."""
-    return collect_logs(args.tarfile, args.userdata, args.verbosity)
+    return collect_logs(
```
Comment (holmanb) [inline cloudinit/cmd/devel/logs.py:330]: was this intended to be in the cleanup commit?

**Review by holmanb — COMMENTED** [review]
@TheRealFalcon the PR looks good, but I think some of the changes meant for the fix PR ended up in the cleanup PR, and tests fail on the first commit:
```
FAILED tests/unittests/cmd/devel/test_logs.py::TestCollectInstallerLogs::test_include_installer_logs_when_present[no_files_include_userdata] - AttributeError: module 'cloudinit.cmd.devel.logs' has no attribute '_collect_installer_logs'. Did you mean: 'collect_installer_logs'?
FAILED tests/unittests/cmd/devel/test_logs.py::TestCollectInstallerLogs::test_include_installer_logs_when_present[no_files_exclude_userdata] - AttributeError: module 'cloudinit.cmd.devel.logs' has no attribute '_collect_installer_logs'. Did you mean: 'collect_installer_logs'?
FAILED tests/unittests/cmd/devel/test_logs.py::TestCollectInstallerLogs::test_include_installer_logs_when_present[files_and_dirs_include_userdata] - AttributeError: module 'cloudinit.cmd.devel.logs' has no attribute '_collect_installer_logs'. Did you mean: 'collect_installer_logs'?
FAILED tests/unittests/cmd/devel/test_logs.py::TestCollectInstallerLogs::test_include_installer_logs_when_present[files_and_dirs_exclude_userdata] - AttributeError: module 'cloudinit.cmd.devel.logs' has no attribute '_collect_installer_logs'. Did you mean: 'collect_installer_logs'?
```

**Comment by TheRealFalcon** [comment]
I had to rebase because local linting was acting weird, so the most recent changes might not be obvious. I'll explain them here.

For your two comments of `was this intended to be in the cleanup commit?`, the answer is no actually. The first commit had initially also moved arg parsing around, but I later noticed it broke some other code so I undid most of it, but the named arguments stuck around. But I agree that they make more sense in the other commit, so I moved them.

Beyond those two, the other change was the test_logs.py change of moving the `logs.collect_installer_logs` -> `logs._collect_installer_logs` to the later commit, allowing unit tests to pass cleanly on both commits.

I also amended the commit message of both commits to include the PR number here.

**Review by holmanb — APPROVED** [review]
Nit: I would use `fix` rather than `refactor` for the first commit, since it makes functional changes which resolve undesirable behavior. Otherwise this looks good to me!

**Pull request merged**